enhancements/storage/csi-driver-install.md

```diff
@@ -30,110 +30,342 @@ superseded-by:
+  this enhancement.
+* `IBMCloud`: While adding a CSI driver for IBM cloud should be trivial and follow concepts outlined in this enhancement,
+  we do not have access to this cloud now. Patches are welcome.
 
 ## Goals
 
 We would like to approach various goals of this KEP in different phases, because we do not expect to get everything right or done in phase-1.
 
 ### Phase-1 Goals
 
-* Provide a way for AWS and GCE CSI driver installation . Users could either optionally install it or CSI driver could be installed
-along with in-tree driver and users could use both.
+* Provide a way for AWS and Manila CSI driver installation. Users could either optionally install it or CSI driver could be installed
+  along with in-tree driver and users could use both.
 * Install CSI provided storageclass along with in-tree StorageClass.
 
 ### Phase-2 Goals
 
-* Support installation of GCE, Cinder, vSphere, AzureDisk CSI drivers.
+* Support installation of oVirt, GCE, Cinder, vSphere, AzureDisk CSI drivers.
+  * Note that exact list of cloud CSI drivers is subject of change during each OCP release planning.
 * Provide a way for users to enable in-tree to CSI migration. It basically replaces whole in-tree volume plugin with a less tested CSI implementation. While it's not a strict requirement, we'd like to have the migration optional for at least one release.
 * Provide a way for users to disable in-tree to CSI migration.
 
 ### Phase-3 Goals
 
-* Enable CSI drivers as default drivers and CSI provided storageclass as default storageclass.
+* Enable all cloud-specific CSI drivers as default drivers and CSI provided storageclass as default storageclass.
 * Optionally allow users to configure CSI driver install.
 
 ## Non-Goals
 
-* This KEP does not attempt to design installation of third-party or partner CSI drivers.
+* This KEP does not attempt to design installation of third-party, partner CSI drivers or non-cloud CSI drivers provided
+  by Red Hat (such as Ceph for OCS).
 * Design of CSI migration enablement.
   * A new operator will be needed to enable alpha features on API server, controller-manager and kubelets in the right order and with proper node draining before switching the gates.
   * Right now it's implementation detail, this enhancement focuses on CSI driver installation. Proper enhancement will be created for CSI migration.
+* Include IBMCloud CSI driver. It can be added later when requested and we have access to the cloud.
 
 ## Proposal
 
-We are currently considering using OLM for installation of CSI driver.
+We are currently considering using CVO and cluster-storage-operator (CSO) for installation of cloud CSI drivers.
+
+Each CSI driver is deployed via a dedicated operator (i.e. each CSI driver has its own), to accommodate differences
+between the CSI drivers. For example, Manila CSI driver operator can detect if Manila service is present in underlying
+OpenStack cloud and deploy the driver only when the service is present.
```

Comment:
This says `Manialla CSI driver` can detect if manilla service is present but section below says CSO detects cloudprovider.  So what happens when CSO installs manila operator but the manila operator determines if manila service is not present?

Same thing applies to vsphere CSI driver too.

Reply:
The Manila operator should advertise in its CR.Status.Conditions that everything is ok (nothing is degraded), but it won't run any Manila CSI driver, because Manila is not there. CSO should translate it into "Available: true", but with a note that only e.g. Cinder CSI driver is running.